api/include/opentelemetry/logs/logger_provider.h

```diff
@@ -44,4 +45,29 @@ class LoggerProvider
 };
 }  // namespace logs
 OPENTELEMETRY_END_NAMESPACE
-#endif
+
+#  if defined(OPENTELEMETRY_EXPORT)
+
+namespace std
+{
+
+//
+// Partial specialization of default_delete used by unique_ptr.
+// This makes the delete of the type in unique_ptr happening in the DLL where it
+// is allocated.
+//
+template <>
+class OPENTELEMETRY_EXPORT default_delete<OPENTELEMETRY_NAMESPACE::logs::LoggerProvider>
+{
+public:
+  void operator()(OPENTELEMETRY_NAMESPACE::logs::LoggerProvider *logger_provider)
+  {
+    delete logger_provider;
+  }
+};
+
+}  // namespace std
+
+#  endif  // defined(OPENTELEMETRY_EXPORT)
```

Comment:
`default_delete<OPENTELEMETRY_NAMESPACE::logs::LoggerProvider>` is only implemented on windows, when `OPENTELEMETRY_EXPORT` is defined.

Assuming the default implementation used by unique_ptr / shared_ptr does the same anyway to delete an object, should be define `default_delete<OPENTELEMETRY_NAMESPACE::logs::LoggerProvider>` on all platforms, to have less differences between windows and non windows code ?

Reply:
The customized deletor is for separating the `delete` into DLL, other platforms without this issue don't need this customization which could incur non-necessary perf overhead. I don't assume the default is implemented like this so apply this to all platforms.

Reply:
Ok, thanks for the clarification.